holidays/countries/japan.py

```diff
@@ -143,166 +143,22 @@ def _populate(self, year):
-            (3, 21, {1988, 2005, 2016, 2033, 2044, 2050}),
-            (3, 22, {1982, 1999, 2010, 2027}),
-            (
-                4,
-                30,
-                {
-                    1973,
-                    1979,
-                    1984,
-                    1990,
-                    2001,
-                    2007,
-                    2012,
-                    2018,
-                    2029,
-                    2035,
-                    2040,
-                    2046,
-                },
-            ),
-            (5, 4, {1981, 1987, 1992, 1998}),
-            (
-                5,
-                6,
-                {
-                    1974,
-                    1985,
-                    1991,
-                    1996,
-                    2002,
-                    2013,
-                    2019,
-                    2024,
-                    2030,
-                    2041,
-                    2047,
-                    2008,
-                    2014,
-                    2025,
-                    2031,
-                    2036,
-                    2042,
-                    2009,
-                    2015,
-                    2020,
-                    2026,
-                    2037,
-                    2043,
-                    2048,
-                },
-            ),
-            (7, 21, {1997}),
-            (8, 9, {2021}),
-            (8, 12, {2019, 2024, 2030, 2041, 2047}),
-            (9, 16, {1974, 1985, 1991, 1996, 2002}),
-            (9, 23, {2024}),
-            (9, 24, {1973, 1984, 1990, 2001, 2007, 2018, 2029, 2035, 2046}),
-            (10, 11, {1976, 1982, 1993, 1999}),
-            (
-                11,
-                4,
-                {
-                    1974,
-                    1985,
-                    1991,
-                    1996,
-                    2002,
-                    2013,
-                    2019,
-                    2024,
-                    2030,
-                    2041,
-                    2047,
-                },
-            ),
-            (
-                11,
-                24,
-                {
-                    1975,
-                    1980,
-                    1986,
-                    1997,
-                    2003,
-                    2008,
-                    2014,
-                    2025,
-                    2031,
-                    2036,
-                    2042,
-                },
-            ),
-            (12, 24, {1990, 2001, 2007, 2012, 2018}),
-        )
-        for holiday in table:
-            month = holiday[0]
-            day = holiday[1]
-            years = holiday[2]
-            if year in years:
-                self[date(year, month, day)] = "振替休日"
+        # When a national holiday falls on Sunday, next working day
+        # shall become a public holiday (振替休日) - substitute holidays
+        for dt in sorted(list(self.keys())):
```

Comment:
I'm not sure if you need `sorted()` here and below. But if you do, the `for dt in sorted(self.keys()):` would work just fine.

Reply:
Yes, `sorted()` isn't needed here. Thank you!